presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkShufflePageInput.java

```diff
@@ -76,32 +78,76 @@ public Page getNextPage()
             while (currentIteratorIndex < shuffleInputs.size()) {
                 PrestoSparkShuffleInput input = shuffleInputs.get(currentIteratorIndex);
                 Iterator<Tuple2<MutablePartitionId, PrestoSparkMutableRow>> iterator = input.getIterator();
-                long processedBytes = 0;
+                long currentIteratorProcessedBytes = 0;
+                long currentIteratorProcessedRows = 0;
+                long currentIteratorProcessedEntries = 0;
                 long start = System.currentTimeMillis();
                 while (iterator.hasNext() && output.size() <= TARGET_SIZE && rowCount <= MAX_ROWS_PER_PAGE) {
+                    currentIteratorProcessedEntries++;
                     PrestoSparkMutableRow row = iterator.next()._2;
                     if (row.getBuffer() != null) {
                         ByteBuffer buffer = row.getBuffer();
-                        output.writeBytes(buffer.array(), buffer.arrayOffset() + buffer.position(), buffer.remaining());
-                        processedBytes += buffer.remaining();
+                        verify(buffer.remaining() >= 1, "row must contain at least a single byte");
```

Comment:
lol

Reply:
Now even a zero columns row will contain at least a single byte, that is a marker.